### PR TITLE
[Test] Optimize test_trtllm_gen_fused_moe.py

### PR DESCRIPTION
**Pull request by jiahanc**
## 📌 Description
Currently `test_llm_gen_fused_moe` took long time, make some optimization to speed up
- Add autotuner choice to turn on/off autotuner in test
- Optimize token count for better coverage
- Optimize `check_accuracy` to speed up

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [ ] Tests have been added or updated as needed.
- [ ] All tests are passing (`unittest`, etc.).

## Summary by CodeRabbit

* **Tests**
  * Add configurable autotune flag (enabled/disabled) across MoE test paths and parametrized cases; propagate it through runtime/test flows
  * Ensure autotune-aware kernel invocations respect a max-token tuning limit (4096)
  * Clear tuner state between runs to improve cache isolation
* **Bug Fixes**
  * Tighten accuracy checks to enforce finiteness and use isclose-based matching with early exit when thresholds met

### CODE REVIEW

File: tests/moe/test_trtllm_gen_fused_moe.py [hunk 1]
```diff
@@ -2267,7 +2282,7 @@ def run_moe_test(
 
 
 # Test: Renormalize routing
-@pytest.mark.parametrize("num_tokens", [1, 8, 1024, 3072])
+@pytest.mark.parametrize("num_tokens", [8, 768, 3072])
```
Comment (jiahanc) [inline tests/moe/test_trtllm_gen_fused_moe.py:2286]: token 1 and 8 uses same kernel, test 8 only. Replace 1024 to 768 to speed up while using same kernel

File: tests/moe/test_trtllm_gen_fused_moe.py [hunk 2]
```diff
@@ -2586,7 +2605,7 @@ def test_topk_routing(
 
 
 # Test: Llama4 routing
-@pytest.mark.parametrize("num_tokens", [1, 8, 1024])
+@pytest.mark.parametrize("num_tokens", [8, 768, 3072])
```
Comment (yzh119) [inline tests/moe/test_trtllm_gen_fused_moe.py]: ditto

File: tests/moe/test_trtllm_gen_fused_moe.py [hunk 3]
```diff
@@ -1408,20 +1418,18 @@ def routing_reference_topk(expert_logits, top_k, num_experts, padding):
...
     """Unified accuracy checking function with detailed error reporting."""
-    if torch.any(torch.isnan(a)):
-        raise Exception("NaN in reference output")
-    if torch.any(torch.isnan(b)):
-        raise Exception("NaN in actual output")
-    if torch.any(torch.isinf(a)):
-        raise Exception("Inf in reference output")
-    if torch.any(torch.isinf(b)):
-        raise Exception("Inf in actual output")
+    if not torch.isfinite(a).all():
```
Comment (yzh119) [inline tests/moe/test_trtllm_gen_fused_moe.py:1425]: It should be equivalent to `torch.any(torch.isinf(a))`, do you observe any performance difference?

File: tests/moe/test_trtllm_gen_fused_moe.py [hunk 4]
```diff
@@ -207,6 +208,7 @@ def _run_moe_computation(self, runtime_args):
             routing_method_type=self.config["routing_method_type"],
             gated_act_type=self.config["gated_act_type"],
             do_finalize=True,
+            tune_max_num_tokens=4096,
```
[the remark this answers is not in the document]
Comment (yzh119) [reply tests/moe/test_trtllm_gen_fused_moe.py]: I think Gemini's suggestion is fair.

File: tests/moe/test_trtllm_gen_fused_moe.py [hunk 5]
```diff
@@ -2295,8 +2307,9 @@ def run_moe_test(
                 "routing_method_type": RoutingMethodType.Renormalize,
                 "compatible_moe_impls": [FP8BlockScaleMoe, FP4Moe, BF16Moe],
                 "compatible_intermediate_size": [384, 768, 1024],
+                "enable_autotune": True,
```
Comment (yzh119) [inline tests/moe/test_trtllm_gen_fused_moe.py:2320]: Are there any cases where this argument is not set to True in this unittest?

File: tests/moe/test_trtllm_gen_fused_moe.py (hunk 1 above)
Comment (yzh119) [reply tests/moe/test_trtllm_gen_fused_moe.py:2286]: thanks for clarification

File: tests/moe/test_trtllm_gen_fused_moe.py (hunk 3 above)
Comment (jiahanc) [reply tests/moe/test_trtllm_gen_fused_moe.py:1425]: There are 2 separate call `torch.isnan` and `torch.isinf` to check if there is nan or inf in tensor. This could be done in 1 `torch.isfinite(a).all()`

File: tests/moe/test_trtllm_gen_fused_moe.py (hunk 5 above)
Comment (jiahanc) [reply tests/moe/test_trtllm_gen_fused_moe.py:2320]: Yes, there are 2 test config set to false with 2 reasons (DSLite and Renorm)
1. Want to test the accuracy without autotune
2. The config is not a real world model config, so should be safe to disable autotune

Comment (yzh119) [reply tests/moe/test_trtllm_gen_fused_moe.py:2320]: thanks for explaination

**Review by yzh119 — APPROVED** [review]
